simplejson/encoder.py

```diff
@@ -39,6 +39,22 @@ def _import_speedups():
 
 FLOAT_REPR = repr
 
+class _JsonStr(str):
```

Comment:
I like the idea of RawJSON and avoiding the copy. To do this, would you then suggesting change `encoder_listencode_obj` to check for this new type, ensure the proper encoding for `encoded_json` and copy the string to a new value? I didn't do this originally because I didn't want to duplicate those portions of `escape_ascii_unicode` and `escape_ascii_str`, but if there is a simple way to avoid that I'm happy to rework this.